### PR TITLE
EventManager 0.2

### PR DESCRIPTION
**Pull request by aljesusg**

### CODE REVIEW

File: app/models/manageiq/providers/consumption/event_manager.rb [hunk 1]
```diff
@@ -13,29 +13,22 @@ def self.update_events
     events = case Time.now.utc.strftime("%d").to_i
              when 1
                generate_new_month
-             else events_month
+             else ManageIQ::Providers::Consumption::ConsumptionManager::ShowbackEvent.events_actual_month
```
Comment (sergio-ocon) [inline app/models/manageiq/providers/consumption/event_manager.rb]: ::ShowbackEvent.events_actual_month???

File: app/models/manageiq/providers/consumption/event_manager.rb [hunk 2]
```diff
@@ -13,29 +13,22 @@ def self.update_events
...
     events.each do |event|
       event.update
     end
   end
 
-  def self.events_month(s_time = DateTime.now.utc.beginning_of_month, e_time =  DateTime.now.utc.end_of_month)
-    ShowbackEvent.where("start_time >= ? AND end_time <= ?",s_time, e_time)
-  end
-
   def self.init_month
```
Comment (sergio-ocon) [inline app/models/manageiq/providers/consumption/event_manager.rb]: Let's call it as:
start_of_month and end_of_month

The way it is written right now looks like you are going to get the first month of something (i.e. January)

File: app/models/manageiq/providers/consumption/event_manager.rb [hunk 3]
```diff
@@ -13,29 +13,22 @@ def self.update_events
...
-  def self.end_month
-    DateTime.now.utc.end_of_month
-  end
-
   def self.generate_new_month
-     events = events_month(DateTime.now.utc.beginning_of_month - 1.months, DateTime.now.utc.end_of_month - 1.months)
+     events = ManageIQ::Providers::Consumption::ConsumptionManager::ShowbackEvent.events_past_month
      events.each do |ev|
-      ShowbackEvent.new({
+       next unless !ShowbackEvent.where(["start_time >= ?",self.init_month]).exists?({:resource_id => ev.resource_id,:resource_type => ev.resource_type})
```
Comment (sergio-ocon) [inline app/models/manageiq/providers/consumption/event_manager.rb]: next if ? (without the unless !)

File: app/models/manageiq/providers/consumption/event_manager.rb (hunk 2 above)
Comment (aljesusg) [inline app/models/manageiq/providers/consumption/event_manager.rb]: Ok I'll add this too

File: app/models/manageiq/providers/consumption/event_manager.rb (hunk 3 above)
Comment (aljesusg) [inline app/models/manageiq/providers/consumption/event_manager.rb]: Rubocop rules

File: app/models/manageiq/providers/consumption/event_manager.rb (hunk 1 above)
Comment (aljesusg) [inline app/models/manageiq/providers/consumption/event_manager.rb]: Yeahh, fast code = not clean code

**Pull request merged**